engine/src/main/java/org/terasology/rendering/animation/ColorInterpolator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.animation;
+
+import org.terasology.rendering.nui.Color;
+
+public abstract class ColorInterpolator implements Frame.FrameComponentInterface {
```

Comment:
I'd rather cut this PR back and complete it with correct implementations of fewer animators. If you insist, you might want to convert the colors into HSB and interpolate there. The RGB space isn't suited for this task.